deploy/role.yaml

```diff
@@ -1,30 +1,51 @@
 apiVersion: rbac.authorization.k8s.io/v1
-kind: ClusterRole
+kind: Role
```

Comment:
This is referenced in install instructions, therefore this change might be a breaking change for consumers of this file.

The instructions encourage to deploy Jaeger in `observability` namespace so with this change the operator will be monitoring only that namespace.

Reply:
I have some thoughts here. The OLM and probably a set users are fine to install the operator with cluster-wide permissions so providing a `ClusterRole` is in place.

In fact `RoleBinding` can reference `ClusterRole`. So we can provide keep this fine in the original form and then have two bindings: `RoleBinding` and `ClusterRoleBinding`.

https://kubernetes.io/docs/reference/access-authn-authz/rbac/#rolebinding-and-clusterrolebinding

> A RoleBinding may also reference a ClusterRole to grant the permissions to namespaced resources defined in the ClusterRole within the RoleBinding’s namespace. This allows administrators to define a set of common roles for the entire cluster, then reuse them within multiple namespaces.

Reply:
True, this raises the question about what the default value for `WATCH_NAMESPACE` should be. On one hand, "secure by default" sounds good and would allow people to try the operator out even on very restricted setups. On the other hand, most people would arguably not take advantage of the operator at its fullest, and might wonder why specific things aren't working, like inject sidecars based on namespace annotations.

cc @objectiser

Reply:
For some reason, your second comment didn't show up when I wrote the previous comment. Interesting find about RoleBinding with ClusterRole. I'll give it a try.

Reply:
The latest version of this PR uses an expanded version of the approach you suggested, by creating two cluster roles instead of a regular role. 

Then, regular role bindings are used for namespaced resources, and an extra cluster role binding file exists, to create the connection between the wider cluster role + the service account. Note that the cluster role binding ships with a default namespace (`observability`), which has to be manually changed by the user in case they want to use a different namespace. This is pretty much the same as it was before, except that this is now the only file with an explicit namespace.

Reply:
I think I need to revert this, at least for now. `operator-sdk generate csv` gets `ClusterRole` copied to its clusterPermission section, and `Role` into role:

https://github.com/operator-framework/operator-sdk/blob/master/doc/user/olm-catalog/generating-a-csv.md#configuration

https://github.com/jaegertracing/jaeger-operator/blob/fd64f15cdd13bbe8360ca9fad2178a06838e3101/deploy/olm-catalog/jaeger-operator/1.16.0/jaeger-operator.v1.16.0.clusterserviceversion.yaml#L125-L220

https://github.com/jaegertracing/jaeger-operator/blob/fd64f15cdd13bbe8360ca9fad2178a06838e3101/deploy/olm-catalog/jaeger-operator/1.16.0/jaeger-operator.v1.16.0.clusterserviceversion.yaml#L260-L355